src/utils/dev.js

```diff
@@ -96,77 +124,61 @@ const getSiteInformation = async ({ flags = {}, api, site, warn, error: failAndE
   return { addonsUrls: {}, teamEnv: {}, addonsEnv: {}, siteEnv: {}, dotFilesEnv, siteUrl: '', capabilities: {} }
 }
 
-// Convenience method for logging a message when an environment variable is overridden by another source (parent) with
-// a higher precedence.
-const logIgnoredEnvVar = ({ key, log, parentSource, source }) =>
-  log(
-    chalk.dim(
-      `${NETLIFYDEVLOG} Ignored ${chalk.bold(source)} env var: ${chalk.yellow(key)} (defined in ${parentSource})`,
-    ),
-  )
-
-const addEnvVariables = ({ log, teamEnv, addonsEnv, siteEnv, dotFilesEnv }) => {
-  const environment = new Map()
-
-  for (const { file, env } of dotFilesEnv) {
-    for (const key in env) {
-      const source = chalk.green.bold(`${file} file`)
-
-      if (environment.has(key)) {
-        logIgnoredEnvVar({ key, log, parentSource: environment.get(key).source, source })
-      } else {
-        environment.set(key, { source, value: env[key] })
-      }
-    }
-  }
+const getEnvSourceName = (source) => {
+  const { printFn = chalk.green, name = source } = ENV_VAR_SOURCES[source] || {}
 
-  for (const key in siteEnv) {
-    const source = chalk.blue.bold('build setting')
+  return printFn(name)
+}
 
-    if (environment.has(key)) {
-      logIgnoredEnvVar({ key, log, parentSource: environment.get(key).source, source })
-    } else {
-      environment.set(key, { source, value: siteEnv[key] })
+// Takes a set of environment variables in the format provided by @netlify/config, augments it with variables from both
+// dot-env files and the process itself, and injects into `process.env`.
+const injectEnvVariables = async ({ env, log, site, warn }) => {
+  const environment = new Map(Object.entries(env))
+  const dotEnvFiles = await loadDotEnvFiles({ projectDir: site.root, warn })
+
+  for (const { file, env: fileEnv } of dotEnvFiles.reverse()) {
```

Comment:
I think `utils/dev.js` is the only consumer for `loadDotEnvFiles`.
If that's the case should we return the result in `loadDotEnvFiles` in the correct order instead of reversing it?

Reply:
Good point. I'll update.